lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1155,6 +1164,45 @@ findFieldWithName(const std::vector<swift::reflection::FieldInfo> &fields,
   return child_indexes.size();
 }
 
+static llvm::Optional<std::string>
+GetMultiPayloadEnumCaseName(const swift::reflection::EnumTypeInfo *eti,
+                            const DataExtractor &data) {
+  assert(eti->getEnumKind() == swift::reflection::EnumKind::MultiPayloadEnum);
+  auto payload_capacity = eti->getPayloadSize();
+  if (data.GetByteSize() == payload_capacity + 1) {
+    auto tag = data.GetDataStart()[payload_capacity];
+    const auto &cases = eti->getCases();
+    if (tag >= 0 && tag < cases.size())
+      return cases[tag].Name;
+  }
+  return {};
+}
+
+llvm::Optional<std::string> SwiftLanguageRuntimeImpl::GetEnumCaseName(
+    CompilerType type, const DataExtractor &data, ExecutionContext *exe_ctx) {
+  using namespace swift::reflection;
+  using namespace swift::remote;
+  auto *ti = GetTypeInfo(type, exe_ctx->GetFramePtr());
+  if (ti->getKind() != TypeInfoKind::Enum)
+    return {};
+
+  auto *eti = llvm::cast<EnumTypeInfo>(ti);
+  PushLocalBuffer((int64_t)data.GetDataStart(), data.GetByteSize());
+  auto defer = llvm::make_scope_exit([&] { PopLocalBuffer(); });
+  RemoteAddress addr(data.GetDataStart());
+  int case_index;
+  if (eti->projectEnumValue(*GetMemoryReader(), addr, &case_index))
+    return eti->getCases()[case_index].Name;
+
+  // Temporary workaround.
+  if (eti->getEnumKind() == EnumKind::MultiPayloadEnum &&
+      type.GetMangledTypeName().GetStringRef().startswith(
+          "$s10Foundation9IndexPathV7Storage10"))
+    return GetMultiPayloadEnumCaseName(eti, data);
```

Comment:
The work around is currently limited to `Foundation.IndexPath.Storage`, because that's the only case surfaced by the lldb test suite. I'd like this to be exhaustive, but I'll have to do some work to identify other any other multipayload enums in Foundation that we need to test and hardcode a special case for.